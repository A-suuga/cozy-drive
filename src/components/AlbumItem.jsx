import styles from '../styles/albumsList'

import React, { Component } from 'react'
// import classNames from 'classnames'
import { Link, withRouter } from 'react-router'
import { translate } from '../lib/I18n'

import { getPhotoLink } from '../actions/photos'

export class AlbumItem extends Component {
  constructor (props) {
    super(props)

    this.state = {
      isLoading: true
    }

    if (props.album && props.album.photosIds.length) {
      getPhotoLink(props.album.photosIds[0])
        .then(link => this.setState({
          url: link,
          isLoading: false
        }))
    } else {
      this.setState({
        url: '',
        isLoading: false
      })
    }
  }

  render () {
    const { t, album, router } = this.props
    const { isLoading, url } = this.state
    const parentPath = router.location.pathname
    return (
      !isLoading &&
        <div className={styles['pho-album']}>
          <Link
            to={`${parentPath}/${album._id}`}
            className={styles['pho-album-link']}
          >
            <img
              className={styles['pho-album-photo-item']}
              src={url}
            />
            <h2 className={styles['pho-album-title']}>{album.name}</h2>
            <h4 className={styles['pho-album-description']}>
              {t('Albums.album_item_description',
                {smart_count: album.photosIds.length})
              }
            </h4>
          </Link>
        </div>
    )
  }
}

export default translate()(withRouter(AlbumItem))
