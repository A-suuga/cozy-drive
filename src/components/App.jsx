import styles from '../styles/app'

import React from 'react'
import { translate } from '../lib/I18n'
import classNames from 'classnames'

import Sidebar from './Sidebar'

const App = ({ t, children }) => (
  <div className={classNames(styles['pho-wrapper'], styles['coz-sticky'])}>

    <Sidebar />

    <main className={styles['pho-content']}>
      { children }
    </main>
  </div>
)

export default translate()(App)
