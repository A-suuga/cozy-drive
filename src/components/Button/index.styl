@require 'cozy-ui'

@media (max-width: (1023/basefont)rem)
    .dri-btn--more
        display inline-flex

@media (max-width: (768/basefont)rem)
    .dri-btn--more
        border        0
        margin-right  0
        height        em(48px, 14px)
        opacity       .5