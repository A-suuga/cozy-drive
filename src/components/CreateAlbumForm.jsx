import styles from '../styles/createAlbumForm'

import React, { Component } from 'react'
import { translate } from '../lib/I18n'

import classNames from 'classNames'

class CreateAlbumForm extends Component {
  constructor () {
    super()
    this.reset()
  }

  reset () {
    this.setState({
      name: '',
      isSubmitDisabled: true,
      isBusy: false,
      hasError: false
    })
  }

  onSubmit (event, callback) {
    event.preventDefault()
    this.setState({ isBusy: true })
    return callback(this.state.name)
      .then(
        () => this.reset(),
        () => this.setState({
          hasError: true,
          isBusy: false
        })
      )
  }

  onNameChange (event) {
    const name = event.target.value.trim()
    this.setState({
      name: name,
      isSubmitDisabled: !name.length,
      hasError: false
    })
  }

  render ({ t, onSubmitNewAlbum }) {
    return (
      <form onSubmit={(event) => this.onSubmit(event, onSubmitNewAlbum)}>
        <label className={styles['coz-create-album-label']}>
          {t('Albums.create.inline_form.create_label')}
        </label>
        <div className={styles['coz-inline-form']}>
          <input
            className={classNames(styles['coz-input-text'], this.state.hasError && styles['error'])}
            type='text'
            name='album-name'
            onInput={(event) => this.onNameChange(event)}
            disabled={this.state.isBusy}
            placeholder={t('Albums.create.inline_form.placeholder')}
            value={this.state.name}
            />
          <button
            className={classNames('coz-btn', 'coz-btn--regular', styles['coz-btn'])}
            disabled={this.state.isSubmitDisabled || this.state.isBusy}
            aria-busy={this.state.isBusy}
            >
            {t('Albums.create.inline_form.create_button')}
          </button>
        </div>
      </form>
    )
  }
}

export default translate()(CreateAlbumForm)
