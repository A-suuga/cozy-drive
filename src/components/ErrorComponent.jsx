import styles from '../styles/errorComponent'

import React from 'react'
import { translate } from '../lib/I18n'

export const ErrorComponent = ({ t, errorType }) => {
  return (
    <div>
      {errorType === 'albums' &&
        <div className={styles['pho-error']}>
          <h2>{ t('Error.albums_title') }</h2>
          <button
            role='button'
            className='coz-btn coz-btn--regular'
            onClick={() => window.location.reload()}
          >
            {t('Error.refresh')}
          </button>
        </div>
      }
    </div>
  )
}

export default translate()(ErrorComponent)
