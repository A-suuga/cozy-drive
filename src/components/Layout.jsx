import styles from '../styles/layout'

import React from 'react'
import classNames from 'classnames'
import { translate } from '../lib/I18n'

import Sidebar from './Sidebar'
import Alerter from './Alerter'

export const Layout = ({ t, children }) => (
  <div className={classNames(styles['pho-wrapper'], styles['coz-sticky'])}>
    <Sidebar />

    <main className={styles['pho-content']}>
      { children }
    </main>
    <Alerter t={t} />
  </div>
)

export default translate()(Layout)
