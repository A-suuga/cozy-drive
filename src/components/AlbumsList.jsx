import styles from '../styles/albumsList'

import React from 'react'

import Empty from '../components/Empty'
import AlbumItem from '../components/AlbumItem'

export const AlbumsList = ({ t, f, albums, onServerError }) => {
  if (albums.length === 0) {
    return <Empty emptyType='albums' />
  } else {
    return (
      <div className={styles['pho-album-list']}>
        {albums.map((a) => <AlbumItem album={a} key={a._id} onServerError={onServerError} />)}
      </div>
    )
  }
}

export default AlbumsList
