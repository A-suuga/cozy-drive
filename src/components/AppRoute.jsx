import React from 'react'
import { Route, Redirect } from 'react-router'

import Timeline from '../containers/Timeline'
import AlbumsView from '../containers/AlbumsView'
import AlbumPhotos from '../containers/AlbumPhotos'
import Viewer from '../containers/Viewer'

import App from './App'
export const ComingSoon = () => (<p style='margin-left: 2em'>Coming soon!</p>)

const AppRoute = (
  <Route component={App}>
    <Redirect from='/' to='photos' />
    <Route path='photos' component={Timeline}>
      <Route path=':photoId' component={Viewer} />
    </Route>
    <Route path='albums' component={AlbumsView}>
      <Route path=':albumId' component={AlbumPhotos}>
        <Route path=':photoId' component={Viewer} />
      </Route>
    </Route>
  </Route>
)

export default AppRoute
