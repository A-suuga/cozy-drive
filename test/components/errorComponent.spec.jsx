'use strict'

/* eslint-env jest */

import React from 'react'
import { shallow } from 'enzyme'

import { mockT } from '../lib/I18n'
import { ErrorComponent } from '../../src/components/ErrorComponent'

describe('Empty component', () => {
  beforeEach(() => {
    jest.resetModules()
  })

  it('should be displayed with photos text if errorType is photos', () => {
    const component = shallow(
      <ErrorComponent t={mockT} errorType='photos' />
    ).node
    expect(component).toMatchSnapshot()
  })

  it('should be displayed with albums text if errorType is albums', () => {
    const component = shallow(
      <ErrorComponent t={mockT} errorType='albums' />
    ).node
    expect(component).toMatchSnapshot()
  })
})
