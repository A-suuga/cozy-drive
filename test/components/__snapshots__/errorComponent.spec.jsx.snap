exports[`Empty component should be displayed with albums text if errorType is albums 1`] = `
<div>
  <div
    className="pho-error">
    <h2>
      Something went wrong while fetching the albums.
    </h2>
    <button
      className="coz-btn coz-btn--regular"
      onClick={[Function]}
      role="button">
      Refresh now
    </button>
  </div>
</div>
`;

exports[`Empty component should be displayed with photos text if errorType is photos 1`] = `<div />`;
